### PR TITLE
Throw err if getter's is used before server init

### PR DESCRIPTION
**Pull request by tamil777selvan**
## Proposed changes

Proposal for #11208 

Throw an error when attempting to call `getValue`, `getValueFromPool`, `addValueToPool` before the server is initialised.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

File: packages/wdio-shared-store-service/tests/client.test.ts [hunk 1]
```diff
@@ -41,6 +41,12 @@ describe('client', () => {
     })
 
     describe('when used in launcher process', () => {
+        it('Error should be thrown when attempting to retrieve a value before server initialization', () => {
+            expect(getValue('*')).rejects.toThrowError('Attempting to use `getValue` before the server has been initialized.')
+            expect(getValueFromPool('*')).rejects.toThrowError('Attempting to use `getValueFromPool` before the server has been initialized.')
+            expect(addValueToPool('*', '')).rejects.toThrowError('Attempting to use `addValueToPool` before the server has been initialized.')
```
Comment (erwinheitzman) [inline packages/wdio-shared-store-service/tests/client.test.ts:47]: This is not retrieving, put it in a separate testcase?

**Review by erwinheitzman — APPROVED** [review]
Approved but I do have a suggestion for the testcase

File: packages/wdio-shared-store-service/tests/client.test.ts (hunk 1 above)
Comment (tamil777selvan) [reply packages/wdio-shared-store-service/tests/client.test.ts:47]: Considering that these are negative tests and should run before invoking `setPort(port)`, which is responsible for resolving the value of `isBaseUrlReady`, I've made this changes. Please share your feedback on this. If necessary, I'm open to move this to a new test case.

Comment (christian-bromann) [inline packages/wdio-shared-store-service/tests/client.test.ts:47]: I agree, this should be:
```suggestion
            expect(getValueFromPool('*', '')).rejects.toThrowError('Attempting to use `addValueToPool` before the server has been initialized.')
```

**Review by christian-bromann — APPROVED** [review]
👍

File: packages/wdio-shared-store-service/tests/client.test.ts (hunk 1 above)
Comment (christian-bromann) [reply packages/wdio-shared-store-service/tests/client.test.ts:47]: I think it makes sense this way.

**Pull request merged**